First Commit

zh4ng opened the pull request:
The footprint belongs to a TI Fuel Gauge - bq27441-G1. Maybe the name
must be changed or the part name must added. I think ;)

zh4ng committed:
First Commit

The footprint belongs to a TI Fuel Gauge - bq27441-G1. Maybe the name
must be changed or the part name must added. I think ;)

zh4ng committed:
KLC Adjustments - VSON-12

KLC 10.7 / 6.5 / 6.9

zh4ng commented:
@SchrodingersGat  Question about Rule 10.7 I used the changes to adjust the solderpaste layer. Is it ok?

SchrodingersGat commented:
Hi @zh4ng generally pretty good, a few things outlined below.

Regarding 10.7 yes you can ignore this because you are meeting the datasheet specification.

### Naming
As this is a very specific footprint, please use the following naming scheme: `Manufacturer_PN`.

In this case, `Texas_S-PDSO-N12`

### Tags
The description tags you have used are for the function of the schematic symbol, not the footprint. Please change the tags to describe the footprint e.g. "SON thermal pads" or similar

### Rotation
Please rotate the part such that pin-1 is at the top left as is the standard

### Pin-1 Marking
For IC devices, pin-1 should be marked on the F.Fab layer by a 45 degree bevel (shown in example image below). Also the silkscreen marking for pin 1 should match other devices e.g. a straight line as shown.

![fp](https://cloud.githubusercontent.com/assets/10080325/24021594/d5afbbf8-0af5-11e7-9b09-78a012515ffb.png)

(It's helpful to look at how other footprints look in the library, this will give you a good guide)

### Pad shape
It looks like one of the pads used to mark the stencil paste is the wrong size:
![wrong_pad](https://cloud.githubusercontent.com/assets/10080325/24021588/d1bb480a-0af5-11e7-8e46-b535db5aa1d8.png)

zh4ng committed:
Merge remote-tracking branch 'refs/remotes/KiCad/master'

zh4ng commented:
Hi @SchrodingersGat 
Thank you for your effort and patience. I have fixed the issues. Bevor is push the next version, i have an other question.
I renamed the footprint to Texas_S-PDSO-N12. The filename and 3DShape also are changed. All fine.

![vson_beta](https://cloud.githubusercontent.com/assets/7514507/24070854/525f004e-0bc5-11e7-9093-98d23329d87b.PNG)

I have added the recommanded solder mask clearance of 0.05mm to the fooprints. Is it Ok?

This is my second footprint. My learning curve in Kicad is incredible as former eagle user.

What is about with the corresponding 3DShape repo?

SchrodingersGat commented:
That solder mask clearance should be fine (I will check when you push the new changes).

I do not understand your question regarding 3D repo - can you phrase this differently? :)

zh4ng committed:
KLC Adustments

Rotate Chip etc.

zh4ng committed:
Courtyard adjustments

SchrodingersGat commented:
Two more issues:

1. You should delete the **VSON-** part (still remained after renaming)
1. The footprint is not precisely centered on (0,0). Please move it to (0,0) origin position.

You have done very good work on this footprint. Fix these small issues and I can merge :)

zh4ng commented:
Hi @SchrodingersGat,
i am fighting with git :) 
Yes, i have seen it.
Damn it, i can't center the footprint. Have you any hint?

zh4ng committed:
Remove VSON Version

SchrodingersGat commented:
To fix this easily:

1. Grab the "Future" version of the KLC scripts here - https://github.com/SchrodingersGat/kicad-library-utils 
1. Run `python check_kicad_mod.py Texas_S-PDSO-N12.kicad_mod -r 8.2 --fix`

zh4ng commented:
Hm, it not working under windows. Tomorrow i have a Linux client at work.

![kicadutils](https://cloud.githubusercontent.com/assets/7514507/24081882/3fef9c82-0cbc-11e7-84bd-890cac26a9f6.PNG)

SchrodingersGat commented:
@zh4ng I am using Windows and that worked for me yesterday. What version of Python are you using?

zh4ng commented:
@SchrodingersGat at work the script worked well. I use python 2.7.13. I need to research

SchrodingersGat commented:
Please let me know what you discover

SchrodingersGat commented:
I have fixed the Python2 errors now on the future KLC branch. Thanks for pointing them out.

They should now work on WIndows/Linux on Python2/3 without error

zh4ng commented:
@SchrodingersGat hm, i downloaded the last version from  https://github.com/SchrodingersGat/kicad-library-utils. Same error on two Windows 10 PCs. Python 2.7.13, one PC with fresh a python.

SchrodingersGat commented:
Ah that's my bad, I hadn't merged the latest changes into the master branch. Please try once more - it should work this time.

zh4ng committed:
Merge remote-tracking branch 'refs/remotes/KiCad/master'

zh4ng committed:
KLC Adjustments

Center the anchor...

zh4ng commented:
Ok. A fixed problem is poping off again :( I'll fix it....

zh4ng committed:
Courtyard Adjustments

zh4ng committed:
Courtyard second try

zh4ng commented:
Hm, i found an other bug. The Solderpaste Layer on 4 Pads is not ok. Fix it in the afternoon..... :( This footprint don't like me....

zh4ng commented:
@SchrodingersGat your turn :)

SchrodingersGat commented:
Now the connection pads are missing the solder-paste layer!

![2017-04-01 09_20_35-3d viewer](https://cloud.githubusercontent.com/assets/10080325/24571397/9fe31b3c-16bc-11e7-80b7-2ea57fe9bd99.png)

zh4ng committed:
Fixing SolderPaste Clearance

SchrodingersGat commented:
Great, this looks much better! Thanks for the contribution :)
